src/main/java/com/easytestit/generatexml/GenerateXML.java

```diff
@@ -1,145 +1,163 @@
+    /**
+     * Constructor a {@link GenerateXML()} class with {@link ConfigureXMLReport()} as a parameter
+     * is necessary when you need to use functionality with your specific configuration data.
+     *
+     * @param configuration the parameter which store specific configuration
+     */
+    public GenerateXML(final ConfigureXMLReport configuration) {
+        if (configuration != null)
+            this.configuration = configuration;
     }
 
     /**
-     * <p> Method describes functionality which provided files and generates the aggregated XML report.
+     * This method is the entry point to the application for converting files
+     * from JSON to XML. This method analyzes the configuration, which may have been passed
+     * to the constructor of class {@link GenerateXML()}.
+     *
+     * It is also analyzed depending on the configuration, what will be done.
+     * <p>For example, if the configuration was transferred without creating a
+     * ZIP of the archive, then this functionality {@link com.easytestit.generatexml.service.ZipService()#createZip()}
+     * for creating the archive will not be launched.
+     * <p/>
+     *
      * When generating process fails report with information about error is provided.
-     * </p>
      */
     public void make() {
         //create new folder where new XML report will be created
         createResultsReportFolder();
 
         try {
-            if (configureXMLReport != null) {
-                //read JSON files from compiled directory
-                configureXMLReport.addJsonFiles(getJsonFilesFrom(configureXMLReport.getReportFolder()));
+            if (configuration != null) {
+                if (configuration.getSource() == null) {
+                    configuration.setSource(ConfigDataProvider.DEFAULT_FOLDER);
+                }
+                if (configuration.getProjectName() == null) {
+                    configuration.setProjectName(ConfigDataProvider.PROJECT_NAME);
+                }
                 //convert JSON file in XML
-                new Serialization().serializeToXML(
-                        new TransformService().transformFeaturesToReport(
-                                new ParseJSON(configureXMLReport).parse()));
-                //create ZIP file from XML which created from previews step and send it to report server
-                if (configureXMLReport.containsConfigurationMode(ConfigureXMLMode.ZIP_XML_RESULT_TO_FILE)) createZip();
-                if (configureXMLReport.containsConfigurationMode(ConfigureXMLMode.SEND_RESULT_TO_RP)) sendXML();
+                convert(resolvePath(configuration.getSource()), configuration.getProjectName(), configuration);
+                //create ZIP file from XML which created from previews step
+                boolean isZipCreated = false;
+                if (configuration.isReportAsZip()) {
+                    isZipCreated = configuration.getProjectName() != null
```

Comment:
is there can be null?

Reply:
No. For now, I was made global changes for the configuration and I checked configuration on null on first initialization. So, I hope there isn't be a null value in the projectName variable.